consumer/src/main/java/com/example/consumer/transaction/TransactionListener.java

```diff
@@ -0,0 +1,31 @@
+package com.example.consumer.transaction;
+
+import com.example.consumer.consumer.KafkaConsumerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import static com.example.consumer.consumer.TopicName.TRANSACTIONS;
+
+@Slf4j
+@Component
+public class TransactionListener implements ApplicationListener<ApplicationReadyEvent> {
```

Comment:
same here: instead of listening to the spring boot events, we can use spring-kafka features to start a consumer

Reply:
updated